salt/modules/yumpkg.py

```diff
@@ -145,6 +145,8 @@ def _yum():
         if ('fedora' in __grains__['os'].lower()
            and int(__grains__['osrelease']) >= 22):
             __context__[contextkey] = 'dnf'
+        elif (__grains__['os_family'] == 'RedHat'
+             and int(__grains__['osrelease']) >= 8):
```

Comment:
You're definitely missing something here - `__context__['contextkey'] = '???'`?

Reply:
We're in yumpkg.py so os_family check is not required. With the following patch, pkg.list_holds works as well for me... 

--- yumpkg.old.py       2020-03-20 13:30:13.220915885 -0500
+++ yumpkg.new.py       2020-03-20 13:29:52.182414799 -0500
@@ -147,6 +147,9 @@
         if ('fedora' in __grains__['os'].lower()
            and int(__grains__['osrelease']) >= 22):
             __context__[contextkey] = 'dnf'
+        elif ('centos' in __grains__['os'].lower()
+           and int(__grains__['osmajorrelease']) >= 8):
+           __context__[contextkey] = 'dnf'
         else:
             __context__[contextkey] = 'yum'
     return __context__[contextkey]
@@ -212,7 +215,8 @@
     Ensure that the appropriate versionlock plugin is present
     '''
     if _yum() == 'dnf':
-        if int(__grains__.get('osmajorrelease')) >= 26:
+        if ('fedora' in __grains__['os'].lower() and int(__grains__.get('osmajorrelease')) >= 26) or \
+           ('centos' in __grains__['os'].lower() and int(__grains__.get('osmajorrelease')) >= 8):
             if six.PY3:
                 vl_plugin = 'python3-dnf-plugin-versionlock'
             else: